src/drivers/fluid_adriver.c

```diff
@@ -476,49 +434,42 @@ delete_fluid_audio_driver(fluid_audio_driver_t* driver)
  */
 int fluid_audio_driver_register(const char** adrivers)
 {
-    unsigned int i=0, add=0;
-    int res = FLUID_FAILED;
+    unsigned int i;
+    uint8_t      disable_mask[(FLUID_N_ELEMENTS(fluid_audio_drivers)+7)/8];
     
-    if(adrivers == NULL)
-    {
-        res = FLUID_OK;
-        goto cleanup;
-    }
-    else
+    if (adrivers == NULL)
+      return FLUID_OK;
```

Comment:
From the doc:

`@param adrivers [...] Pass NULL to register all available drivers.`